client/state/data-layer/wpcom/jetpack/settings/index.js

```diff
@@ -30,7 +35,11 @@ export const fromApi = response => {
 		throw new Error( 'missing settings' );
 	}
 
-	return response.data;
+	return normalizeSettings( response.data );
+};
+
+export const toApi = settings => {
```

Comment:
The `toApi` helper and the rest of the changes here could use some unit tests.

Reply:
Hmm, `toApi` is the composition of two reasonably well tested util functions. A _unit_ test (in the proper meaning of the word) would actually have to verify that (for arbitrary `settings`), `toApi( settings ) === filterSettingsByActiveModules( sanitizeSettings( settings ) );`. I'm tempted to skip 😁

Reply:
Fine for the `toApi` (although I'd probably add a simple test just to ensure that it's doing what it's expected), but what about the rest of the changes?